packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -693,8 +693,11 @@ class ReactDOMServerRenderer {
     for (let i = this.providerIndex; i >= 0; i--) {
       // We assume this Flow type is correct because of the index check above.
       const priorProvider: ?ReactProvider<any> = this.providerStack[i];
-      if (priorProvider !== null && priorProvider !== undefined &&
-        priorProvider.type === provider.type) {
+      if (
+        priorProvider !== null &&
+        priorProvider !== undefined &&
```

Comment:
These checks aren’t truly necessary, are they? We know they can’t be empty because we’re moving to the left and check the index.

If type system disagrees I think we should do an unsafe cast with a comment.

Reply:
Happy to. I wasn't sure how strict you wanted to be with the type checking.